guides/cheatsheets/axon_pytorch.cheatmd

```diff
@@ -0,0 +1,546 @@
+# Axon -> PyTorch
+
+This cheatsheet is designed to assist PyTorch developers in transitioning to Elixir and Axon,
+providing equivalent commands and code examples for common neural network tasks.
+
+## Core Paradigm: Functional vs. Object-Oriented
+
+A key difference between Axon and PyTorch lies in their core design paradigms:
+
+### Axon (Functional)
+Axon follows a functional approach, inspired by libraries like JAX.
+Models are defined as compositions of functions that transform input data and parameters into output data.
+State (like model parameters) is managed explicitly and passed into functions.
+This promotes purity, explicit data flow, and composability, often leveraging Just-In-Time (JIT) compilation
+via Nx backends and compilers (like EXLA) for performance.
+
+### PyTorch (Object-Oriented)
+PyTorch uses an object-oriented approach.
+Models are typically defined as classes inheriting from `torch.nn.Module`.
+These classes encapsulate layers and parameters as internal state.
+The forward pass is defined as a method (`forward`) operating on this internal state.
+This provides a familiar structure for many developers but can sometimes obscure
+data flow and state management compared to the functional style.
+
+This cheatsheet will highlight how common tasks are achieved in both paradigms.
+
+## Model Definition
+
+### Sequential Models
+
+#### Axon
+
+```elixir
+model =
+  Axon.input("input", shape: {nil, 784})
+  |> Axon.dense(128, activation: :relu)
+  |> Axon.dense(10, activation: :softmax)
+```
+
+#### PyTorch
+
+```python
+import torch
+import torch.nn as nn
+
+model = nn.Sequential(
+    nn.Linear(784, 128),
+    nn.ReLU(),
+    nn.Linear(128, 10),
+    nn.Softmax(dim=1)
+)
+```
+
+## Common Layer Types
+
+### Dense / Linear
+
+Applies a linear transformation to the incoming data: `y = xW^T + b`.
+
+#### Axon
+```elixir
+Axon.dense(out_features, activation: :relu, name: "my_dense_layer")
+dense_layer = Axon.dense(128)
+```
```

Comment:
This needs the inputs! The graph always has to be built in a "connected" way:

```suggestion
```elixir
input = Axon.input("features")
Axon.dense(input, out_features, activation: :relu, name: "my_dense_layer")
dense_layer = Axon.dense(input, 128)
```
```

Reply:
You can mention if you want to do the same thing as in pytorch you can use `Axon.block`. That will turn a sub-graph into a re-usable function

Reply:
There are few times where I want to react with a 🤦‍♂️ emoji.

This is one of those times.